.test-infra/jenkins/job_PerformanceTests_PubsubIO_Python.groovy

```diff
@@ -0,0 +1,106 @@
+ * limitations under the License.
+ */
+
+import CommonJobProperties as commonJobProperties
+import LoadTestsBuilder as loadTestsBuilder
+import PhraseTriggeringPostCommitBuilder
+
+import static java.util.UUID.randomUUID
+
+def now = new Date().format("MMddHHmmss", TimeZone.getTimeZone('UTC'))
+def timeout = 60 * 60
+
+def pubsubId = randomUUID()
+def pubsubName = "pubsub_io_performance_${pubsubId}"
+
+def psio_write_test = [
+        title          : 'PubsubIO Write Performance Test Python 2GB',
+        test           : 'apache_beam.io.gcp.pubsub_write_perf_test',
+        runner         : CommonTestProperties.Runner.DATAFLOW,
+        pipelineOptions: [
+                job_name             : 'performance-tests-psio-write-python-2gb' + now,
+                project              : 'apache-beam-testing',
+                temp_location        : 'gs://temp-storage-for-perf-tests/loadtests',
+                publish_to_big_query : true,
+                metrics_dataset      : 'beam_performance',
+                metrics_table        : 'psio_write_2GB_msg_results',
+                input_options        : '\'{' +
+                        '"num_records": 2097152,' +
+                        '"key_size": 1,' +
+                        '"value_size": 1024}\'',
+                num_workers          : 5,
+                autoscaling_algorithm: 'NONE',  // Disable autoscale the worker pool.
+                pubsub_name          : pubsubName,
+                timeout              : timeout,
+        ]
+]
+
+def psio_read_test = [
+        title          : 'PubsubIO Read Performance Test Python 2GB',
+        test           : 'apache_beam.io.gcp.pubsub_read_perf_test',
+        runner         : CommonTestProperties.Runner.DATAFLOW,
+        pipelineOptions: [
+                job_name             : 'performance-tests-psio-read-python-2gb' + now,
+                project              : 'apache-beam-testing',
+                temp_location        : 'gs://temp-storage-for-perf-tests/loadtests',
+                publish_to_big_query : true,
+                metrics_dataset      : 'beam_performance',
+                metrics_table        : 'psio_read_2GB_msg_results',
+                input_options        : '\'{' +
+                        '"num_records": 2097152,' +
+                        '"key_size": 1,' +
+                        '"value_size": 1024}\'',
+                num_workers          : 5,
+                autoscaling_algorithm: 'NONE',  // Disable autoscale the worker pool.
+                pubsub_name          : pubsubName,
+                timeout              : timeout,
+        ]
+]
+
+def executeJob = { scope, testConfig ->
+    commonJobProperties.setTopLevelMainJobProperties(scope, 'master', 240)
+
+    loadTestsBuilder.loadTest(scope, testConfig.title, testConfig.runner,
+            CommonTestProperties.SDK.PYTHON_37, testConfig.pipelineOptions, testConfig.test)
+}
+
+PhraseTriggeringPostCommitBuilder.postCommitJob(
```

Comment:
I like the idea of reusing output topic for the purpose of the read test. But, in that case, you have to make sure that the write test is executed first. At the moment, both tests start independently at the same hour ('H 15 * * *'), which can cause some issues.

Reply:
I moved both tests to one file so now it'll be impossible to run them separately. They were so tightly bounded that it made no sense to run them individually